crates/transaction-emitter-lib/src/emitter/local_account_generator.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+use aptos_sdk::types::{AccountKey, LocalAccount};
+use aptos_transaction_generator_lib::{AccountType, ReliableTransactionSubmitter};
+use async_trait::async_trait;
+use futures::future::try_join_all;
+use rand::rngs::StdRng;
+
+#[async_trait]
+pub trait LocalAccountGenerator: Send + Sync {
+    async fn gen_local_accounts(
+        &self,
+        txn_executor: &dyn ReliableTransactionSubmitter,
+        num_accounts: usize,
+        rng: &mut StdRng,
+    ) -> anyhow::Result<Vec<LocalAccount>>;
+}
+
+pub fn create_account_generator(account_type: AccountType) -> Box<dyn LocalAccountGenerator> {
+    match account_type {
+        AccountType::Local => Box::new(PrivateKeyAccountGenerator),
+        _ => {
+            unimplemented!("Account type {:?} is not supported", account_type)
+        },
+    }
+}
+
+pub struct PrivateKeyAccountGenerator;
+
+#[async_trait]
+impl LocalAccountGenerator for PrivateKeyAccountGenerator {
+    async fn gen_local_accounts(
+        &self,
+        txn_executor: &dyn ReliableTransactionSubmitter,
+        num_accounts: usize,
+        rng: &mut StdRng,
+    ) -> anyhow::Result<Vec<LocalAccount>> {
+        let mut account_keys = vec![];
+        let mut addresses = vec![];
+        let mut i = 0;
+        while i < num_accounts {
+            let account_key = AccountKey::generate(rng);
+            addresses.push(account_key.authentication_key().account_address());
+            account_keys.push(account_key);
```

Comment:
should the method here be just create_local_account (with seq_num=0), and then in the minter do the batch seq_num fetch and update  - as that part will be the same for all types of accounts?

Reply:
Hmm.. I think the batch API might be needed for keyless as we will need to read a file and create the accounts. In any case, I can iterate on it once I understand how to add keyless account support.

Reply:
You can read file in memory, and have a counter that gives you the next identity.

The point is that batch network communication to fetch sequence numbers is the same,  and should live outside - generation should just provide account initialized with sequence number = 0

Reply:
Sure, let me evolve in the next PR which adds the keyless account.